build.assets/build-fido2.sh

```diff
@@ -0,0 +1,209 @@
+#!/bin/bash
+#
+# Builds libfido2 and dependencies, caching the resulting binaries in the local
+# filesystem.
+#
+# Run `build-fido2.sh build` to build libfido2 and its dependencies, at the
+# versions specified in the script.
+# Run `build-fido2.sh pkg_config_path` to print the path to the prior-built
+# libfido2-static.pc file.
+#
+# Written mainly for macOS builders.
+set -eu
+
+# Note: versions are the same as the corresponding git tags for each repo.
+readonly CBOR_VERSION=v0.9.0
+readonly CRYPTO_VERSION=OpenSSL_1_1_1o
+readonly FIDO2_VERSION=1.11.0
+
+readonly LIB_CACHE="$HOME/.teleport-fido2-cache"
+
+readonly PKGFILE_DIR="$LIB_CACHE/fido2-${FIDO2_VERSION}_cbor-${CBOR_VERSION}_crypto-${CRYPTO_VERSION}"
+
+fetch_and_build() {
+  local name="$1"      # eg, cbor
+  local version="$2"   # eg, v0.9.0
+  local url="$3"       # eg, https://github.com/...
+  local buildcmd="$4"  # eg, cbor_build, a bash function name
+  echo "$name: fetch and build" >&2
+
+  local tmp=''
+  tmp="$(mktemp -d)"
+  # Early expansion on purpose.
+  #shellcheck disable=SC2064
+  trap "rm -fr '$tmp'" exit
+
+  local fullname="$name-$version"
+  local install_path="$tmp/$fullname"
```

Comment:
Have you considered having `install_path` be a tmpdir in `LIB_CACHE` so the final install is atomic?